### DISCUSSION TITLE
Generate Cloudflare _redirects via adapter

### DISCUSSION THREAD

**Discussion by oliwarner** [withastro/roadmap#642] [Proposal]
### Body

# Summary

The new redirects framework works well in static sites but only generates files with meta-refresh tags. Server-side redirects are quicker. Cloudflare supports a `_redirects` file that is very similar to Netlifly's in syntax and the Netlifly adapter already does this. This should be an easy port.

# Goals

- ~~Generate _redirects from known redirects - similar to `createRedirects` in Netlifly's integration, and generate a file `_redirects`~~ (it already does this - it just can't run in static mode)
- Allow the Cloudflare adapter to be enabled in `mode: "static"`

**Comment by florian-lefebvre** [comment]
Maybe you can rename your proposal so that it accounts for any adapter (eg vercel and netlify). I think the meta-refresh tag hurts seo (see https://discord.com/channels/830184174198718474/1139314258144940122/1139863743682060398)

**Comment by oliwarner** [comment]
Well this is galling. The Cloudflare adaptor [already has code that does everything I'm requesting](https://github.com/withastro/astro/blob/5c23bf1c901006ea9615827430642c2041a6cf8d/packages/integrations/cloudflare/src/index.ts#L362), it's just blocked from running from by [a mode == static check](https://github.com/withastro/astro/blob/5c23bf1c901006ea9615827430642c2041a6cf8d/packages/integrations/cloudflare/src/index.ts#L107).

I've tried nuking the check in my local copy but the code is so intertwined with functions handlers, it still fails when it can't find functions.
